staticaddr/manager.go

```diff
@@ -0,0 +1,188 @@
+package staticaddr
+
+import (
+	"context"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/staticaddr/script"
+	"github.com/lightninglabs/loop/swap"
+	staticaddressrpc "github.com/lightninglabs/loop/swapserverrpc"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// ManagerConfig holds the configuration for the address manager.
+type ManagerConfig struct {
+	// AddressClient is the client that communicates with the loop server
+	// to manage static addresses.
+	AddressClient staticaddressrpc.StaticAddressServerClient
+
+	// SwapClient provides loop rpc functionality.
+	SwapClient *loop.Client
+
+	// Store is the database store that is used to store static address
+	// related records.
+	Store AddressStore
+
+	// WalletKit is the wallet client that is used to derive new keys from
+	// lnd's wallet.
+	WalletKit lndclient.WalletKitClient
+
+	// ChainParams is the chain configuration(mainnet, testnet...) this
+	// manager uses.
+	ChainParams *chaincfg.Params
+}
+
+// Manager manages the address state machines.
+type Manager struct {
+	cfg *ManagerConfig
+
+	initChan chan struct{}
+
+	sync.Mutex
+}
+
+// NewAddressManager creates a new address manager.
+func NewAddressManager(cfg *ManagerConfig) *Manager {
+	return &Manager{
+		cfg:      cfg,
+		initChan: make(chan struct{}),
+	}
+}
+
+// Run runs the address manager.
+func (m *Manager) Run(ctx context.Context) error {
+	log.Debugf("Starting address manager.")
+	defer log.Debugf("Address manager stopped.")
+
+	// Communicate to the caller that the address manager has completed its
+	// initialization.
+	close(m.initChan)
+
+	<-ctx.Done()
+
+	return nil
+}
+
+// NewAddress starts a new address creation flow.
+func (m *Manager) NewAddress(ctx context.Context) (*btcutil.AddressTaproot,
+	error) {
+
+	// If there's already a static address in the database, we can return
+	// it.
+	m.Lock()
+	addresses, err := m.cfg.Store.GetAllStaticAddresses(ctx)
+	if err != nil {
+		m.Unlock()
```

Comment:
Can `m.Unlock()` call be done right after `GetAllStaticAddresses` call? In all 3 branches the first action is `m.Unlock()`.